src/transport/PeerConnections.h

```diff
@@ -184,31 +198,37 @@ class PeerConnections
      * @param nodeId is the connection to find (based on nodeId). Note that initial connections
      *        do not have a node id set. Use this if you know the node id should be set.
      * @param peerKeyId Encryption key ID used by the peer node.
-     * @param state [out] the connection if found, null otherwise. MUST not be null.
+     * @param begin If a member of the pool, will start search from the next item. Can be nullptr to search from start.
      *
-     * @return true if a corresponding state was found.
+     * @return the state found, nullptr if not found
      */
     CHECK_RETURN_VALUE
-    bool FindPeerConnectionState(Optional<NodeId> nodeId, uint16_t peerKeyId, PeerConnectionState ** state)
+    PeerConnectionState * FindPeerConnectionState(Optional<NodeId> nodeId, uint16_t peerKeyId, PeerConnectionState * begin)
     {
-        *state = nullptr;
-        for (size_t i = 0; i < kMaxConnectionCount; i++)
+        PeerConnectionState * state = nullptr;
+        PeerConnectionState * now   = &mStates[0];
```

Comment:
I wouldn't use `now` as this suggests time. Maybe use `peer` and rename `state` to `matchingPeer`.